core/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsResponse.java

```diff
@@ -80,66 +81,46 @@ public ClusterStatsIndices getIndicesStats() {
     }
 
     @Override
-    public ClusterStatsNodeResponse[] getNodes() {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public Map<String, ClusterStatsNodeResponse> getNodesMap() {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public ClusterStatsNodeResponse getAt(int position) {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public Iterator<ClusterStatsNodeResponse> iterator() {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
     public void readFrom(StreamInput in) throws IOException {
         super.readFrom(in);
         timestamp = in.readVLong();
-        status = null;
-        if (in.readBoolean()) {
-            // it may be that the master switched on us while doing the operation. In this case the status may be null.
-            status = ClusterHealthStatus.fromValue(in.readByte());
-        }
         clusterUUID = in.readString();
-        nodesStats = new ClusterStatsNodes(in);
-        indicesStats = ClusterStatsIndices.readIndicesStats(in);
+        // it may be that the master switched on us while doing the operation. In this case the status may be null.
+        status = in.readBoolean() ? ClusterHealthStatus.fromValue(in.readByte()) : null;
+        nodes = in.readList(ClusterStatsNodeResponse::readNodeResponse);
+
+        // built from nodes rather than from the stream directly
+        nodesStats = new ClusterStatsNodes(nodes);
+        indicesStats = new ClusterStatsIndices(nodes);
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
         out.writeVLong(timestamp);
+        out.writeString(clusterUUID);
+
         if (status == null) {
             out.writeBoolean(false);
         } else {
             out.writeBoolean(true);
             out.writeByte(status.value());
         }
-        out.writeString(clusterUUID);
-        nodesStats.writeTo(out);
-        indicesStats.writeTo(out);
+
+        out.writeStreamableList(nodes);
     }
 
     static final class Fields {
         static final String NODES = "nodes";
```

Comment:
I believe it is fine to just remove this class and use the strings. That is what we've been doing with new code. If we want field constants the Fields class is really nice. But I don't think the constants buy us much.